src/middlewares.js

```diff
@@ -384,8 +384,19 @@ export function allowCrossDomain(appId) {
     if (config && config.allowHeaders) {
       allowHeaders += `, ${config.allowHeaders.join(', ')}`;
     }
-    const allowOrigin = (config && config.allowOrigin) || '*';
-    res.header('Access-Control-Allow-Origin', allowOrigin);
+
+    // Support for multiple origins
+    let allowedOrigins = config && config.allowOrigin ? config.allowOrigin : ['*'];
+
+    // Convert comma-separated string to an array if needed
+    if (typeof allowedOrigins === 'string') {
+      allowedOrigins = allowedOrigins.split(',').map(domain => domain.trim());
+    }
+
+    const requestOrigin = req.headers.origin;
+    const originToSet =
+      requestOrigin && allowedOrigins.includes(requestOrigin) ? requestOrigin : allowedOrigins[0];
+    res.header('Access-Control-Allow-Origin', originToSet);
```

Comment:
Simplified (since we don't allow comma separated strings) and to keep semantics similar to `allowHeaders`:

```suggestion
    const baseOrigins = config?.allowOrigin ?? ['*'];
    const requestOrigin = req.headers.origin;
    const allowOrigins = requestOrigin && baseOrigins.includes(requestOrigin) ? requestOrigin : baseOrigins[0];
    res.header('Access-Control-Allow-Origin', allowOrigins);
```

Reply:
I had to slightly modify this for the case of a it being a single origin string

`const baseOrigins =
      typeof config?.allowOrigin === 'string' ? [config.allowOrigin] : config?.allowOrigin ?? ['*'];`